### PR TITLE
Support completion for alias and sub-command

### PR DESCRIPTION
**Pull request by Kangaxx-0**
# Description

- Address #5666 , completion could work for arbitrary alias and sub-comamnds
- Fix unit tests

## Demo
`g = git` and `co = checkout`

alias
---
No breaking changes to other completions

image

wala

# Tests

Make sure you've run and fixed any issues with these commands:

- [x] `cargo fmt --all -- --check` to check standard code formatting (`cargo fmt --all` applies these changes)
- [x] `cargo clippy --workspace --features=extra -- -D warnings -D clippy::unwrap_used -A clippy::needless_collect` to check that you're using the standard code style
- [x] `cargo test --workspace --features=extra` to check that all the tests pass

### CODE REVIEW

**Comment by japui** [comment]
5749
Abrir
[Kangaxx-0](https://github.com/Kangaxx-0) quer mesclar 10 commits

**Comment by fdncred** [comment]
wait a second, these look like abbreviations. is that right?

**Comment by fdncred** [comment]
I think fish and perhaps some other shells have the thing called abbr or abbreviations. so you define something like this: `abbr gco = git checkout` and then when you're at the prompt you type `gco <tab>` and it expands the abbreviation to `git checkout <completions>`. I saw it within the last month. I'll try to find a link that shows more.

**Comment by fdncred** [comment]
ya, it was fish `abbr`. if you google it you'll find lots of hits https://fishshell.com/docs/current/cmds/abbr.html

**Comment by Kangaxx-0** [comment]
> ya, it was fish `abbr`. if you google it you'll find lots of hits https://fishshell.com/docs/current/cmds/abbr.html

For me, maybe other users, e.g:  #5666, `alias` plays the same role for Nu. If this is not what we want, could you clarify what would be an expected fix for #5666 please ?

screenshot from zsh config
image

**Comment by fdncred** [comment]
alias and abbr are two separate functionalities. This kind of combines them and maybe that's ok. Maybe we need Dan-Gamin's input since it's they're issue.

**Comment by hustcer** [comment]
Wow, It's a amazing feature.

File: crates/nu-cli/src/completions/completer.rs [hunk 1]
```diff
@@ -229,6 +235,74 @@ impl ReedlineCompleter for NuCompleter {
...
+                }
+            }
+        } else {
+            vec_alias.push(name);
+        }
+    }
+
+    // Zip names and alias vectors, the original inputs and its aliases mapping.
+    // e.g:(['g'], ['g','i','t'])
+    let output = vec_names.clone().into_iter().zip(vec_alias).collect();
```
Comment (WindSoilder) [inline crates/nu-cli/src/completions/completer.rs:298]: May be we can just move this `let output = vec_names.clone().into_iter().zip(vec_alias).collect();` under the `if is_alias`?

And I don't think the final clone on vec_names is really needed.

**Comment by fdncred** [comment]
@Kangaxx-0 i'm loving this. thanks for contributing it!

**Comment by fdncred** [comment]
there's some debate whether this is working or not. I can only get it to work with the following.
```
alias g = git
alias co = checkout
> g co <tab> and it expands to git checkout
```
however, if I do this
```
alias k = kubectl
alasi cf = config
> k cf <tab> nothing happens.
```
what am I missing?

**Comment by Kangaxx-0** [comment]
> there's some debate whether this is working or not. I can only get it to work with the following.
> 
> ```
> alias g = git
> alias co = checkout
> > g co <tab> and it expands to git checkout
> ```
> 
> however, if I do this
> 
> ```
> alias k = kubectl
> alasi cf = config
> > k cf <tab> nothing happens.
> ```
> 
> what am I missing?

`g co` works magically out of the box because the default nu config has already exported `git checkout` as external command, if user wants the auto expand works, then they have to export the customized external command, e.g. 

```
alias cg = cargo
alias tt = test
...
export extern "cargo test" []
```
Then `cg tt <tab>` would work. 
cargo

**Comment by Kangaxx-0** [comment]
BTW, I want to mention this - The main thing my pr was trying to fix   ->  `cg = cargo`, and then `cg test` caused nu panicked

**Comment by fdncred** [comment]
I guess we broke this at some point because this doesn't work any longer.
```nushell
# Abbreviations
alias cg = cargo
alias tt = test
export extern "cargo test" []
```
`cg tt <tab>` doesn't expand

**Pull request merged**